operator/cmd/mesh/uninstall.go

```diff
@@ -0,0 +1,200 @@
+		return fmt.Errorf("could not configure logs: %s", err)
+	}
+	client, err := k8sClient(uiArgs.kubeConfigPath, uiArgs.context)
+	if err != nil {
+		return err
+	}
+
+	cache.FlushObjectCaches()
+	opts := &helmreconciler.Options{DryRun: rootArgs.dryRun, Log: l, ProgressLog: progress.NewLog()}
+	h, err := helmreconciler.NewHelmReconciler(client.Controller(), client.RESTConfig(), nil, opts)
+	if err != nil {
+		return fmt.Errorf("failed to create reconciler: %v", err)
+	}
+
+	// If only revision flag is set, we would prune resources by the revision label.
+	// Otherwise we would merge the revision flag and the filename flag and delete resources by generated manifests.
+	if uiArgs.filename == "" {
+		emptyiops := &v1alpha1.IstioOperatorSpec{Profile: "empty", Revision: uiArgs.revision}
+		iop, err := translate.IOPStoIOP(emptyiops, "empty", iopv1alpha1.Namespace(emptyiops))
+		if err != nil {
+			return err
+		}
+		h, err := helmreconciler.NewHelmReconciler(client.Controller(), client.RESTConfig(), iop, opts)
+		if err != nil {
+			return fmt.Errorf("failed to create reconciler: %v", err)
+		}
+		objectsList, tp, err := h.GetPrunedResources(uiArgs.revision, uiArgs.purge)
+		if err != nil {
+			return err
+		}
+		if err := preCheckWarnings(cmd, rootArgs, uiArgs, uiArgs.revision, tp, l); err != nil {
+			return fmt.Errorf("failed to do preuninstall check: %v", err)
+		}
+
+		if err := h.DeleteObjectsList(objectsList); err != nil {
+			return fmt.Errorf("failed to delete control plane resources by revision: %v", err)
+		}
+		opts.ProgressLog.SetState(progress.StateComplete)
+		return nil
+	}
+	manifestMap, iops, err := manifest.GenManifests([]string{uiArgs.filename},
+		applyFlagAliases(uiArgs.set, uiArgs.manifestsPath, uiArgs.revision), uiArgs.force, client.RESTConfig(), l)
+	if err != nil {
+		return err
+	}
+	if err := preCheckWarnings(cmd, rootArgs, uiArgs, iops.Revision, nil, l); err != nil {
+		return fmt.Errorf("failed to do preuninstall check: %v", err)
+	}
+	if err := h.DeleteControlPlaneByManifests(manifestMap, iops.Revision, uiArgs.purge); err != nil {
+		return fmt.Errorf("failed to delete control plane by manifests: %v", err)
+	}
+	opts.ProgressLog.SetState(progress.StateComplete)
+	return nil
+}
+
+// preCheckWarnings checks possible breaking changes and issue warnings to users, it checks the following:
+// 1. checks proxies still pointing to the target control plane revision.
+// 2. lists to be pruned resources if user uninstall by --revision flag.
+func preCheckWarnings(cmd *cobra.Command, rootArgs *rootArgs, uiArgs *uninstallArgs,
+	rev string, resourcesList []string, l *clog.ConsoleLogger) error {
+	pids, err := proxyinfo.GetIDsFromProxyInfo(uiArgs.kubeConfigPath, uiArgs.context, rev, uiArgs.istioNamespace)
+	if err != nil {
+		return err
+	}
+	needConfirmation := false
+	var message string
+	if resourcesList != nil {
+		needConfirmation = true
+		message += fmt.Sprintf("To be pruned resources from the cluster: %s\n",
```

Comment:
Grammar:  Say "The following resources will be pruned from the cluster: %s".  Appearance: Join with `", "` not `","`.

Also, the list of resources is long.  For me it was:

"To be pruned resources from the cluster: HorizontalPodAutoscaler:istio-system:istiod,PodDisruptionBudget:istio-system:istiod,Deployment:istio-system:istiod,Service:istio-system:istiod,ConfigMap:istio-system:istio,ConfigMap:istio-system:istio-sidecar-injector,EnvoyFilter:istio-system:metadata-exchange-1.4,EnvoyFilter:istio-system:metadata-exchange-1.5,EnvoyFilter:istio-system:metadata-exchange-1.6,EnvoyFilter:istio-system:stats-filter-1.4,EnvoyFilter:istio-system:stats-filter-1.5,EnvoyFilter:istio-system:stats-filter-1.6,EnvoyFilter:istio-system:tcp-metadata-exchange-1.5,EnvoyFilter:istio-system:tcp-metadata-exchange-1.6,EnvoyFilter:istio-system:tcp-stats-filter-1.5,EnvoyFilter:istio-system:tcp-stats-filter-1.6,MutatingWebhookConfiguration::istio-sidecar-injector"

I feel it would be better to summarize them.  There might not be time to do that on this PR.  I will approve it without summarization.  But it would be better to say something like "Removing Deployment:istio-system:istiod, 1 HorizontalPodAutoscaler, 1 PodDisruptionBudget, 1 Kubernetes Service, 2 ConfigMaps, 10 Envoy Filters, and 1 MutatingWebhookConfiguration".

When `--purge` is chosen, the list is super-long.  I am not sure we need it.

Reply:
ok I would reformat the output a bit, maybe group by namespace? And for purge, we can just say all Istio resources would be deleted

Reply:
ok now for the case without purge, output would just contains name kind hash, for the case of purge it would just emphasize that all Istio resources would be removed